lib/fastlane/plugin/wpmreleasetoolkit/helper/git_helper.rb

```diff
@@ -15,22 +15,20 @@ module GitHelper
       #         (i.e. a local working copy) or a subdirectory of one.
       #
       def self.is_git_repo?(path: nil)
-        args = ['git']
-
-        unless path.nil?
-          # `path` might represent a file that doesn't yet exist but is in a
-          # valid Git repo. In such case, the `git rev-parse` we're currently
-          # building would fail. To avoid that, let's perform the check on the
-          # first existing ancestor of the given path – which will be the path
-          # itself if it exists.
-          path = first_existing_ancestor_of(path: path)
-          args += ['-C', File.file?(path) ? File.dirname(path) : path]
-        end
+        working_path = path.nil? ? Dir.pwd : first_existing_ancestor_of(path: path)
+        current_dir = File.directory?(working_path) ? working_path : File.dirname(working_path)
 
-        args += ['rev-parse']
+        # Calling `expand_path` to get an absolute path and avoid and endless
+        # loop of adding ".." to the path without ever triggerting the "/"
+        # safeguard.
+        full_path = Pathname(current_dir).expand_path
 
-        Action.sh(args, print_command_output: false) do |status, _, _|
-          status.success?
+        if full_path.children.map { |f| File.basename(f) }.include?('.git')
```

Comment:
1. I thought `children` already only returned the relative path (ie only the file name) not the full path? Which means `File.basename(f)` is probably unnecessary here?
2. Also, isn't there a more appropriate method in `File` or `Dir` to check if a dir contains a file? What about testing `if File.exist?(File.join(full_path, '.git'))` instead? Or even better, `if File.directory?(File.join(full_path, '.git'))` to both check existence AND that it is a directory not a file? Feels less consuming than asking to list all the children of a dir only to check if a single specific child (`.git`) exists 😉

Reply:
Actually I see that you're using `Pathname` to play with the paths (which is great, I love that class instead of using `File.xxx` and `Dir.xxx` class methods! 😍 ), but then if we have a `Pathname`, maybe better (1) create the `Pathname` earlier in order to take advantage of the nicer API of that type early on in the code (2) use the `Pathname` instance methods everywhere from that point in this implementation, instead of having a mix of `Pathname` instance methods and `File.x(f)` global static methods?

This is also true about the test for root folder on line 28 below: `Pathname` has a `root?` instance method that feels better suited than a comparison to a custom `ROOT_FOLDER` constant 😉